stocking/src/main/scala/dev/atedeg/mdm/stocking/Actions.scala

```diff
@@ -0,0 +1,66 @@
+package dev.atedeg.mdm.stocking
+
+import cats.Monad
+import cats.data.{ NonEmptyList, NonEmptySet }
+import cats.syntax.all.*
+
+import dev.atedeg.mdm.products.*
+import dev.atedeg.mdm.products.given
+import dev.atedeg.mdm.stocking.Errors.*
+import dev.atedeg.mdm.stocking.OutgoingEvent.*
+import dev.atedeg.mdm.utils.*
+import dev.atedeg.mdm.utils.given
+import dev.atedeg.mdm.utils.monads.*
+
+/**
+ * Gets how many products are missing from the stock, given the desired stock.
+ */
+def getMissingCountFromProductStock(
+    availableStock: AvailableStock,
+    desiredStock: DesiredStock,
+)(product: Product): MissingQuantity =
+  if availableStock(product).n >= desiredStock(product).n.toNonNegative then MissingQuantity(0)
+  else MissingQuantity(desiredStock(product).n.toNonNegative - availableStock(product).n)
+
+/**
+ * Removes the given quantity of a certain product from the stock, giving the new current stock.
+ */
+def removeFromStock[M[_]: Monad: CanRaise[NotEnoughStock]](
+    stock: AvailableStock,
+)(product: Product, quantity: DesiredQuantity): M[AvailableStock] =
+  (stock(product).n > quantity.n.toNonNegative)
+    .otherwiseRaise(NotEnoughStock(product, quantity, stock(product)): NotEnoughStock)
+    .thenReturn(stock + (product -> AvailableQuantity(stock(product).n - quantity.n.toNonNegative)))
+
+/**
+ * Approves a batch after quality assurance.
+ */
+def approveBatch(batch: Batch.ReadyForQualityAssurance): QualityAssuredBatch.Passed =
+  QualityAssuredBatch.Passed(batch.id, batch.cheeseType)
+
+/**
+ * Rejects a batch after quality assurance.
+ */
+def rejectBatch(batch: Batch.ReadyForQualityAssurance): QualityAssuredBatch.Failed =
+  QualityAssuredBatch.Failed(batch.id, batch.cheeseType)
+
+/**
+ * Labels a product given the [[QualityAssuredBatch.Passed batch]] it comes from and its [[WeightInGrams actual weight]]
```

Comment:
Missing link to `Product` and broken link to `WeightInGrams`
```suggestion
 * Labels a [[Product product]] given the [[QualityAssuredBatch.Passed batch]] it comes from and its [[Grams actual weight]]
```